sklearnex/spmd/linear_model/tests/test_incremental_linear_spmd.py

```diff
@@ -0,0 +1,329 @@
+        hparams_spmd.gpu_macro_block = macro_block
+
+    for i in range(num_blocks):
+        local_dpt_X = _convert_to_dataframe(
+            split_local_X[i], sycl_queue=queue, target_df=dataframe
+        )
+        local_dpt_y = _convert_to_dataframe(
+            split_local_y[i], sycl_queue=queue, target_df=dataframe
+        )
+        inclin_spmd.partial_fit(local_dpt_X, local_dpt_y)
+
+    inclin.fit(dpt_X, dpt_y)
+
+    assert_allclose(inclin.coef_, inclin_spmd.coef_)
+    if fit_intercept:
+        assert_allclose(inclin.intercept_, inclin_spmd.intercept_)
+
+
+@pytest.mark.skipif(
+    not _mpi_libs_and_gpu_available,
+    reason="GPU device and MPI libs required for test",
+)
+@pytest.mark.parametrize(
+    "dataframe,queue",
+    get_dataframes_and_queues(dataframe_filter_="dpnp,dpctl", device_filter_="gpu"),
+)
+@pytest.mark.parametrize("fit_intercept", [True, False])
+@pytest.mark.parametrize("num_samples", [100, 1000])
+@pytest.mark.parametrize("num_features", [5, 10])
+@pytest.mark.parametrize("macro_block", [None, 1024])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+@pytest.mark.mpi
+def test_incremental_linear_regression_fit_spmd_random(
+    dataframe, queue, fit_intercept, num_samples, num_features, macro_block, dtype
+):
+    # Import spmd and non-SPMD algo
+    from sklearnex.linear_model import IncrementalLinearRegression
+    from sklearnex.spmd.linear_model import (
+        IncrementalLinearRegression as IncrementalLinearRegression_SPMD,
+    )
+
+    tol = 2e-4 if dtype == np.float32 else 1e-7
+
+    # Create gold data and process into dpt
+    X_train, X_test, y_train, _ = _generate_regression_data(
+        num_samples, num_features, dtype
+    )
+    dpt_X = _convert_to_dataframe(X_train, sycl_queue=queue, target_df=dataframe)
+    dpt_X_test = _convert_to_dataframe(X_test, sycl_queue=queue, target_df=dataframe)
+    local_X = _get_local_tensor(X_train)
+    local_dpt_X = _convert_to_dataframe(local_X, sycl_queue=queue, target_df=dataframe)
+
+    dpt_y = _convert_to_dataframe(y_train, sycl_queue=queue, target_df=dataframe)
+    local_y = _get_local_tensor(y_train)
+    local_dpt_y = _convert_to_dataframe(local_y, sycl_queue=queue, target_df=dataframe)
+
+    inclin_spmd = IncrementalLinearRegression_SPMD(fit_intercept=fit_intercept)
+    inclin = IncrementalLinearRegression(fit_intercept=fit_intercept)
+
+    if macro_block is not None:
+        hparams = inclin.get_hyperparameters("fit")
+        hparams.cpu_macro_block = macro_block
+        hparams.gpu_macro_block = macro_block
+
+        hparams_spmd = inclin_spmd.get_hyperparameters("fit")
+        hparams_spmd.cpu_macro_block = macro_block
+        hparams_spmd.gpu_macro_block = macro_block
+
+    inclin_spmd.fit(local_dpt_X, local_dpt_y)
+    inclin.fit(dpt_X, dpt_y)
+
+    assert_allclose(inclin.coef_, inclin_spmd.coef_, atol=tol)
+    if fit_intercept:
+        assert_allclose(inclin.intercept_, inclin_spmd.intercept_, atol=tol)
+
```

Comment:
Does this test actually provide coverage for incremental algos or just base batch?

Reply:
fit for incremental algos splits data into batches and then call partial_fit for each batch. it is done like in scikit incrementalPCA. this test provides coverage for spmd version of this method comparing it with regular IncrementalPCA.fit

Reply:
I figured, just couldn't find specifically where that logic was defined